src/CoreBundle/private/library/classes/TCMSTableEditor/TCMSTableWriter.class.php

```diff
@@ -127,19 +135,76 @@ protected function PostSaveHook(&$oFields, &$oPostTable)
     {
         parent::PostSaveHook($oFields, $oPostTable);
 
+        $newTableName = $oPostTable->sqlData['name'];
+
         // reset table list cache
         if (array_key_exists('_listObjCache', $_SESSION)) {
             $_SESSION['_listObjCache'] = array();
         }
 
         $this->getAutoclassesCacheWarmer()->updateTableById($this->sId);
 
-        // check table engine
-        $realEngine = $this->getRealTableEngine($oPostTable->sqlData['name']);
+        $this->adaptTableEngine($oPostTable, $newTableName);
+        $this->adaptTableName($this->sOldTblName, $newTableName);
+        $this->adaptTableComment($newTableName, $this->oldTableComment);
+    }
+
+    /**
+     * @param TCMSRecord $oPostTable
+     * @param string     $tableName
+     *
+     * @throws DBALException
+     */
+    private function adaptTableEngine(&$oPostTable, string $tableName): void
```

Comment:
I'd like to change two things here:
- Move this code to changeTableEngine, so that there is only one method dealing with the table engine (the difference between `adaptTableEngine` and `changeTableEngine` can only be seen by reading the method bodies); I'm quite sure that we never want to have the table using another engine than is saved in the table config, so we should enforce writing both at the same time. This changes the method contract and we should add a small note in the upgrade guide.
- Rename `adaptTableName` and `adaptTableComment` to `changeXyz`. The `change` prefix is way more common.

Reply:
ok